### PR TITLE
feat(plugin-server): add analytics-ingestion and recordings-ingestion modes

### PR DESCRIPTION
**Pull request by hazzadous**
Add two new modes such that we can separate the workloads in k8s

## Changes

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

**Review by tiina303 — APPROVED** [review]
lgtm, though check the test failure first

File: plugin-server/src/capabilities.ts [hunk 1]
```diff
@@ -2,45 +2,74 @@ import { PluginServerCapabilities, PluginsServerConfig } from './types'
...
 export function getPluginServerCapabilities(config: PluginsServerConfig): PluginServerCapabilities {
+    // Given the server config, which is largely determined by environment
+    // variables, determine which workloads/capabilities this instance of the
+    // plugin-server should run.
+    //
+    // Note that historically we have used `PLUGIN_SERVER_MODE` for determining
+    // the workloads we want to run. However, this makes it difficult to roll
+    // out changes in a controlled manner, so instead we are moving towards
+    // removing this abstraction and instead explicitly specifying the workloads
+    // by way of individual environment variables.
```
[the remark this answers is not in the document]
Comment (hazzadous) [reply plugin-server/src/capabilities.ts:13]: The run everything flag is `PLUGIN_SERVER_MODE` being null which is e.g. what it is in dev.

This new env var, if set, takes priority for which ever workload it is related to.

[the remark this answers is not in the document]
Comment (tiina303) [reply plugin-server/src/capabilities.ts]: so we're planning to keep `PLUGIN_SERVER_MODE`? Your comment seemed to indicate we want to be removing it longer term.

(to clarify my question (which I poorly formed): it was about once we are done moving everything over what will things look like)

[the remark this answers is not in the document]
Comment (hazzadous) [reply plugin-server/src/capabilities.ts]: Yeah I got bogged down in how to roll this out, and decided just to drop that for now. 😓

**Pull request merged**